### PR TITLE
[Feature] Packaging qt5

### PR DESCRIPTION
**Pull request by jpfeuffer**
Plus its platform plugins.
Major rework and simplification of all the packaging CMake scripts.
Kept dependencies to a minimum.
Relies on all Win libs to be static now. See current contrib PRs.
Tested on all platforms but due to many switches back and forth there might be little things that were overseen in the end.
Adapted docu accordingly.

### CODE REVIEW

File: cmake/knime_package_support.cmake [hunk 1]
```diff
@@ -184,84 +184,60 @@ add_custom_target(
...
-    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:OpenSwathAlgo> ${PAYLOAD_LIB_PATH}
-    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:SuperHirn> ${PAYLOAD_LIB_PATH}
-  )
-
-  function(copy_library lib target_path)
-    string(REGEX REPLACE "lib$" "dll" target_dll "${lib}")
-    file(TO_NATIVE_PATH "${target_dll}" target_native)
+else()
+  ## Assemble common required libraries for win and lnx
+  ## Note that we do not need the QT plugins or QTGui libraries since we do not include GUI tools here.
```
Comment (timosachsenberg) [inline cmake/knime_package_support.cmake]: question if we want to ship TOPPView in KNIME

**Comment by timosachsenberg** [comment]
great job! Two things we might change:
- remove IDEvalutator as we kind of agreed to remove it
- bundle TOPPView in KNIME

**Comment by jpfeuffer** [comment]
Sure we can think about that. We can remove the qt librarypath config hack. For TOPPView we would need to leave the guy libraries. TOPPView needs a ctd then. Is it using toppbase already? We have to check if it needs to be a bundle or an executable on Mac. Should we do this here or try to merge first?

**Comment by jpfeuffer** [comment]
rebuild jenkins

**Comment by timosachsenberg** [comment]
let's merge first

**Comment by timosachsenberg** [comment]
maybe remove the hack and IDEvaluator if that's fine with you. TOPPView might need some more work

**Comment by jpfeuffer** [comment]
I think finally we are struck by the locale error. We should fix it.
sprintf should not be used.
See #3203

**Comment by hroest** [comment]
>  sprintf should not be used.

what is the alternative, should we use boost::format ?

**Comment by jpfeuffer** [comment]
Good idea, maybe boost::locale::format?
http://www.boost.org/doc/libs/1_56_0/libs/locale/doc/html/group__format.html#gad7914df7b54382c1ad7f5360676fe2e8

**Comment by hroest** [comment]
@jpfeuffer I have fixed this here: #3253 - was this what you had in mind?

**Comment by timosachsenberg** [comment]
@hroest not sure if this solves the problem. From the doc, it seems to use the global local. Maybe we need to set it explicitly to the US local when we create the string?

**Comment by timosachsenberg** [comment]
I merge to reduce the number of open qt5 branches. let's address the locale problem separately

**Pull request merged**